### DISCUSSION TITLE
Line tooltip - hovered value support

### DISCUSSION THREAD

**Discussion by tervay** [f5/unovis#70] [Q&A] [answered]
Unovis doesn't seem to support a tooltip for a Line component rendering the value that the user is hovering over. For example:

```ts
<VisXYContainer height={500}>
  <VisLine
    data={timeline}  // just {x: number, y: number}[]
    x={(d) => d.x}
    y={(d) => d.y}
  />
  <VisAxis type="x" />
  <VisAxis type="y" />
  <VisTooltip
    triggers={{
      [Line.selectors.line]: (d) => console.log(d),  // *
    }}
  />
</VisXYContainer>
```

You'll notice the format of the `d` parameter is:

```ts
{
  defined: boolean;
  visible: boolean;
  values: {
    x: number;
    y: number;
    defined: boolean;
    value: number;
  }[];
}
```

This `values` field is _all_ data on the line graph; I only want the one value that the user is hovering over. Is this supported?

**Comment by rokotyan** [comment, marked as answer]
@tervay Line and Area components don't really have points on them that can be hovered over. I think what you're looking for is a combination of Tooltip and [Crosshair](https://unovis.dev/docs/auxiliary/Crosshair).

Here's an example from the gallery how to set them up:
https://unovis.dev/gallery/view?collection=%20Auxiliary%20Components&title=Stacked%20Bar%20Chart%20with%20Crosshair

Alternatively, you can [combine](https://github.com/f5/unovis/discussions/38) Line with Scatter and use the tooltip on the scatter plot dots.

**Reply by tervay** [reply]
Thanks; I'll give the crosshair example from the gallery a try.
